### ISSUE TITLE
in Crachtlystcs Fatal Exception: java.util.concurrent.RejectedExecutionException 

### ISSUE THREAD

**Issue by yousefbeiruty** [grpc/grpc-java#9275]
```
Fatal Exception: java.util.concurrent.RejectedExecutionException
Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@c1500d8 rejected from java.util.concurrent.ScheduledThreadPoolExecutor@49f43bb[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 18]
java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution (ThreadPoolExecutor.java:2014)

java.util.concurrent.ScheduledThreadPoolExecutor.execute (ScheduledThreadPoolExecutor.java:616)
io.grpc.internal.SerializingExecutor.schedule (SerializingExecutor.java:93)
io.grpc.internal.SerializingExecutor.execute (SerializingExecutor.java:86)
io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl.closedInternal (ClientCallImpl.java:750)
io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl.closed (ClientCallImpl.java:686)
io.grpc.internal.DelayedStream$DelayedStreamListener$5.run (DelayedStream.java:485)
io.grpc.internal.DelayedStream$DelayedStreamListener.delayOrExecute (DelayedStream.java:426)
io.grpc.internal.DelayedStream$DelayedStreamListener.closed (DelayedStream.java:482)
io.grpc.internal.ForwardingClientStreamListener.closed (ForwardingClientStreamListener.java:39)
io.grpc.internal.InternalSubchannel$CallTracingTransport$1$1.closed (InternalSubchannel.java:697)
io.grpc.internal.AbstractClientStream$TransportState.closeListener (AbstractClientStream.java:459)
io.grpc.internal.AbstractClientStream$TransportState.access$400 (AbstractClientStream.java:221)
io.grpc.internal.AbstractClientStream$TransportState$1.run (AbstractClientStream.java:442)
io.grpc.internal.AbstractClientStream$TransportState.deframerClosed (AbstractClientStream.java:278)
io.grpc.internal.Http2ClientStreamTransportState.deframerClosed (Http2ClientStreamTransportState.java:31)
io.grpc.okhttp.OkHttpClientStream$TransportState.deframerClosed (OkHttpClientStream.java:304)
io.grpc.internal.MessageDeframer.close (MessageDeframer.java:233)
io.grpc.internal.MessageDeframer.closeWhenComplete (MessageDeframer.java:191)
io.grpc.internal.AbstractStream$TransportState.closeDeframer (AbstractStream.java:203)
io.grpc.internal.AbstractClientStream$TransportState.transportReportStatus (AbstractClientStream.java:445)
io.grpc.okhttp.OkHttpClientTransport.startGoAway (OkHttpClientTransport.java:887)
io.grpc.okhttp.OkHttpClientTransport.access$1600 (OkHttpClientTransport.java:109)
io.grpc.okhttp.OkHttpClientTransport$ClientFrameHandler.run (OkHttpClientTransport.java:1113)
java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1113)
```

**Comment by ejona86** [comment]
Are you setting `channelBuilder.executor()`? You are most likely shutting down the executor before the channel has terminated.

**Comment by yousefbeiruty** [comment]
Ok for reproduce this issue i don't know really because this is a production app on
Google play store and this crush message appears on Firebase analytics and i cant see it on my device but here are the dependencies with the versions that i am using:-.



   //analytics
      analytics = '19.0.0'
   //perf
     perf = '20.0.1'
    //messaging
    messaging = '22.0.0'
   //crashlytics
    crashlytics = '18.1.0'
    //storage
    firebasestore = '20.0.0'

    implementation "com.google.firebase:firebase-analytics:$analytics"
    implementation "com.google.firebase:firebase-perf:$perf"
    implementation "com.google.firebase:firebase-messaging:$messaging"
    implementation "com.google.firebase:firebase-crashlytics:$crashlytics"
    implementation "com.google.firebase:firebase-storage:$firebasestore"

  /**PubSub Google Cloud */
    //implementation 'androidx.lifecycle:lifecycle-runtime-ktx:2.4.1'

    implementation('com.google.cloud:google-cloud-pubsub:1.105.0') {
        exclude group: 'org.apache.httpcomponents'
        // exclude group:'com.google.protobuf'
    }
    implementation('com.google.firebase:firebase-inappmessaging-display:19.1.5', {
        exclude group: 'io.grpc', module: 'grpc-okhttp'
    })
    implementation "io.grpc:grpc-okhttp:$okhttp"
    implementation "org.jetbrains.anko:anko:$anko"

**Comment by ejona86** [comment]
@yousefbeiruty, are you using gRPC directly? If so, the `ManagedChannelBuilder`/`OkHttpChannelBuilder` configuration is the important part. Specifically a call to `executor()`.

What version of gRPC are you using? You didn't include the `okhttp` variable in the gradle snippet. Based on the stacktrace, it seems you can't be using 1.40 or later. I know there were some fixes in 1.36 that could impact this sort of thing. I suggest upgrading to a recent vension.

**Comment by ejona86** [comment]
No response to provide more information, so closing. This very well may be a firebase client bug. More information can be provided and then we can reopen.
